backend/src/services/integration-auth/integration-sync-secret.ts

```diff
@@ -695,24 +698,61 @@ const syncSecretsAWSSecretManager = async ({
   integration,
   secrets,
   accessId,
-  accessToken
+  accessToken,
+  awsAssumeRoleArn,
+  projectId
 }: {
   integration: TIntegrations;
   secrets: Record<string, { value: string; comment?: string }>;
   accessId: string | null;
   accessToken: string;
+  awsAssumeRoleArn: string | null;
+  projectId?: string;
 }) => {
+  const appCfg = getConfig();
   const metadata = z.record(z.any()).parse(integration.metadata || {});
 
-  if (!accessId) {
-    throw new Error("AWS access ID is required");
+  if (!accessId && !awsAssumeRoleArn) {
+    throw new Error("AWS access ID/AWS Assume Role is required");
+  }
+
+  let accessKeyId = "";
+  let secretAccessKey = "";
+  let sessionToken;
+  if (awsAssumeRoleArn) {
+    if (!appCfg.CLIENT_ID_AWS_INTEGRATION || !appCfg.CLIENT_SECRET_AWS_INTEGRATION)
+      throw new Error("Missing AWS Credentials for Assuming Role. Contact server admin.");
+
+    const client = new STSClient({
+      region: integration.region as string,
+      credentials: {
+        accessKeyId: appCfg.CLIENT_ID_AWS_INTEGRATION,
+        secretAccessKey: appCfg.CLIENT_SECRET_AWS_INTEGRATION
+      }
```

Comment:
for instances deployed to AWS, I think we should provide an option so that they don't have to explicitly define AWS credentials? 

Like, it should attempt to use the credentials provided in the AWS workload instance if not explicitly defined in the env.

Reply:
Yah i can do that. Just remove the check 😅 and that's all i have to do. AWS SDK tries to find all the various possible ways to load the cred by itself

Reply:
Removed the explicit check. AWS SDK will automatically pick it up